src/os/bluestore/BlueStore.cc

```diff
@@ -1521,7 +1523,10 @@ void BlueStore::ExtentMap::reshard(Onode *o)
       dout(20) << __func__ << " old shard end 0x" << std::hex << shard_end
 	       << std::dec << dendl;
     }
-    if (estimate && estimate + extent_avg > target) {
+    // disfavor shard boundaries that span a blob
+    bool would_span = (ep->logical_offset < max_blob_end) || ep->blob_offset;
```

Comment:
@liewegas I don't understand why it would span if ep->blob_offset is non zero ?
Could you also help me understand what is the difference between logical_offset and blob_offset within an extent ?